tests/test_utils.py

```diff
@@ -733,802 +350,272 @@ def test_update_from_copyright_info_empty(self):
-        assert len(test_delta.factors) == 3
+        deltas = test_delta.deltas
+
+        assert deltas[0].score == 70
+        assert len(deltas[0].factors) == 3
         for factor in expected_factors:
-            assert factor in test_delta.factors
+            assert factor in deltas[0].factors
 
     def test_update_from_lic_copy_info_copyright_change_no_license_change(self):
-        test_file_new = models.File({
-            'path':'/test/path.txt',
-            'name': 'path.txt',
-            'sha1': 'a',
-            'original_path': '',
-            "licenses": [
-                {
-                    "key": "gpl-2.0",
-                    "score": 70.0,
-                    "short_name": "GPL 2.0",
-                    "category": "Copyleft"
-                }
-            ],
-            "copyrights": [
-                {
-                    "statements": [
-                        "Copyright (c) 2017-2018 Francois Hennebique and others."
-                    ],
-                    "holders": [
-                        "Francois Hennebique and others."
-                    ]
-                },
-                {
-                    "statements": [
-                        "Copyright (c) 2015 Edouard-Leon Scott de Martinville."
-                    ],
-                    "holders": [
-                        "Edouard-Leon Scott de Martinville."
-                    ]
-                }
-            ]
-        })
-        test_file_old = models.File({
-            'path':'/test/path.txt',
-            'name': 'path.txt',
-            'sha1': 'a_modified',
-            'original_path': '',
-            "licenses": [
-                {
-                    "key": "gpl-2.0",
-                    "score": 70.0,
-                    "short_name": "GPL 2.0",
-                    "category": "Copyleft"
-                }
-            ],
-            "copyrights": [
-                {
-                    "statements": [
-                        "Copyright (c) 2017-2018 Francois Hennebique and others."
-                    ],
-                    "holders": [
-                        "Francois Hennebique and others."
-                    ]
-                }
-            ]
-        })
+        test_file_new = self.get_test_loc('utils/update_from_lic_copy_info_copyright_change_no_license_change_new.json')
+        test_file_old = self.get_test_loc('utils/update_from_lic_copy_info_copyright_change_no_license_change_old.json')
 
-        test_delta = deltacode.Delta(20, test_file_new, test_file_old)
 
-        utils.update_modified_from_license_info(test_delta, unique_categories)
-        utils.update_modified_from_copyright_info(test_delta)
+        test_delta = DeltaCode(test_file_new, test_file_old, {})
 
-        assert test_delta.score == 25
-        assert len(test_delta.factors) == 1
-        assert 'copyright change' in test_delta.factors
+        deltas = test_delta.deltas
+
+        assert deltas[0].score == 25
```

Comment:
It is better to address to assign the first element of `deltas` to a variable and use that rather than repeatedly getting the delta object from the list by index.

``` 
deltas = test_deltas.deltas

# ensure that deltas is not empty
assert deltas

delta = deltas[0]
assert len(delta.factors) == 1
assert 'copyright change' in delta.factors
```

Please update this in other places where this occurs.

Reply:
Yes that's better changing it soon !